x/wasm/types/types.go

```diff
@@ -22,6 +25,109 @@ const (
 	SDKAddrLen = 20
 )
 
+var (
+	_ authztypes.Authorization = &ExecuteContractAuthorization{}
+)
+
+func NewAllowedContract(contractAddr sdk.AccAddress, allowedMessages []string) *AllowedContract {
+	return &AllowedContract{
+		ContractAddress: contractAddr.String(),
+		AllowedMessages: allowedMessages,
+	}
+}
+
+// NewExecuteContractAuthorization creates a new ExecuteContractAuthorization object.
+func NewExecuteContractAuthorization(allowedContracts []*AllowedContract) *ExecuteContractAuthorization {
+	return &ExecuteContractAuthorization{
+		AllowedContracts: allowedContracts,
+	}
+}
+
+// MsgTypeURL implements Authorization.MsgTypeURL.
+func (a ExecuteContractAuthorization) MsgTypeURL() string {
+	return "/cosmos.wasm.base.MsgExecuteContract"
```

Comment:
```suggestion
	return sdk.MsgTypeURL(&MsgExecuteContract{})
```
should be `/cosmwasm.wasm.v1.MsgExecuteContract` , ref [sendAuthz](https://github.com/cosmos/cosmos-sdk/blob/v0.45.3/x/bank/types/send_authorization.go#L22)